docs/docs/external-volumes

```diff
@@ -0,0 +1,79 @@
+--------------------------------------------------------------
+title: Stateful Applications Using Persistent External Volumes
+--------------------------------------------------------------
+
+# Stateful Applications Using Persistent External Volumes
+
+<div class="alert alert-danger" role="alert">
+  <span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span> Adapted in Marathon Version 1.0 <br/>
+  The Persistent Storage functionality is considered beta, so use this feature at your own risk. We might add, change, or delete any functionality described in this document.
+</div>
+
+Marathon applications lose their state when they terminate and are relaunched. In some contexts, for instance, if your application uses MySQL, you’ll want your application to preserve its state. You can use an external storage service, such as Amazon's Elastic Block Store (EBS), to create a persistent volume that follows your application instance.
+
+Using an external storage service allows your apps to be more fault-tolerant. If a host fails, Marathon reschedules it on another host, along with its associated data, without user intervention.
+
+# Specifying an External Volume
+
+If you are running Marathon on DCOS, add the following to the `genconf.yml` file you use during DCOS installation. [Learn more](https://docs.mesosphere.com/concepts/installing/installing-enterprise-edition/configuration-parameters/).
+
+- `rexray_config_method: file`
+
+- `rexray_config_filename: /path/to/rexray.yaml`
+
+## Scaling your App
+
+Apps that use external volumes should only be scaled to a single instance.
+
+If you scale your app down to 0 instances, the volume is unattached from the agent where it was mounted, but it is not deleted. If you scale your app up again, the data that had been associated with it will still be available.
+
+## Create an Application with External Volumes
+
+You specify an external volume in the app definition of your Marathon app. [Learn more about Marathon application definitions](application-basics.html).
+
+```json
+{ 
+  "id": "hello",
+     "cmd": "while [ true ] ; do touch /var/lib/rexray/volumes/test12345/data/hello ; sleep 5 ; done",   
+     "mem": 32,   
+     "cpus": 0.1,   
+     "instances": 1,
+     "env": {
+          "DVDI_VOLUME_DRIVER": "rexray",
+          "DVDI_VOLUME_NAME": "test12345",
+          "DVDI_VOLUME_OPTS":"size=1,volumetype=standard,newfstype=ext4,overwritefs=false"
+     }
+}
+```
+
+In the app definition above, the environment variables (`env`) specify a 1GB volume that will be formatted ext4 once it is created.
+
+- Volumes will always be available under `$REXRAY_HOME/var/lib/rexray/volumes/$DVDI_VOLUME_NAME`. **Note:** This applies only to non-Docker apps; Docker apps can specify the mount point inside the app definition. See [the REX-Ray documentation on data directories](https://rexray.readthedocs.org/en/v0.3.2/user-guide/config/#data-directories) for more information.
```

Comment:
The MESOS container example I pasted in a comment shows how to do this

On Mon, Apr 4, 2016 at 3:18 PM, Branden Rolston <notifications@github.com>
wrote:

> In docs/docs/external-volumes
> <https://github.com/mesosphere/marathon/pull/3607#discussion_r58431700>:
>
> > +  "id": "hello",
> > +     "cmd": "while [ true ] ; do touch /var/lib/rexray/volumes/test12345/data/hello ; sleep 5 ; done",
> > +     "mem": 32,
> > +     "cpus": 0.1,
> > +     "instances": 1,
> > +     "env": {
> > +          "DVDI_VOLUME_DRIVER": "rexray",
> > +          "DVDI_VOLUME_NAME": "test12345",
> > +          "DVDI_VOLUME_OPTS":"size=1,volumetype=standard,newfstype=ext4,overwritefs=false"
> > +     }
> > +}
> > +```
> > +
> > +In the app definition above, the environment variables (`env`) specify a 1GB volume that will be formatted ext4 once it is created.
> > +
> > +- Volumes will always be available under `$REXRAY_HOME/var/lib/rexray/volumes/$DVDI_VOLUME_NAME`. **Note:** This applies only to non-Docker apps; Docker apps can specify the mount point inside the app definition. See [the REX-Ray documentation on data directories](https://rexray.readthedocs.org/en/v0.3.2/user-guide/config/#data-directories) for more information.
>
> @jdef <https://github.com/jdef> can you clarify? AFAIK the only way to
> get REX-Ray to mount volumes under /tmp is to set REXRAY_HOME=/tmp.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/marathon/pull/3607/files/5de0b7fdac69848ac2ba5442bcb2b72d28935246#r58431700>
>